starlite/plugins/sql_alchemy/config.py

```diff
@@ -0,0 +1,218 @@
+    from starlite.types import Scope
+
+T = TypeVar("T", Session, AsyncSession)
+
+IsolationLevel = Literal["AUTOCOMMIT", "READ COMMITTED", "READ UNCOMMITTED", "REPEATABLE READ", "SERIALIZABLE"]
+
+
+def serializer(value: Any) -> str:
+    """Serializer for JSON field values.
+
+    Args:
+        value: Any json serializable value.
+
+    Returns:
+        JSON string.
+    """
+    return dumps(
+        value,
+        default=default_serializer,
+        option=OPT_SERIALIZE_NUMPY,
+    ).decode("utf-8")
+
+
+class SQLAlchemySessionConfig(BaseModel):
+    class Config(BaseConfig):
+        arbitrary_types_allowed = True
+
+    bind: Optional[Any] = None
+    autoflush: Optional[bool] = None
+    future: Optional[bool] = None
+    autocommit: Optional[bool] = None
+    expire_on_commit: bool = False
+    twophase: Optional[bool] = None
+    binds: Optional[Any] = None
+    enable_baked_queries: Optional[bool] = None
+    info: Optional[Dict[str, Any]] = None
+    query_cls: Optional[Type[Query]] = None
+
+
+class SQLAlchemyEngineConfig(BaseModel):
+    class Config(BaseConfig):
+        arbitrary_types_allowed = True
+
+    connect_args: Optional[Dict[str, Any]] = None
+    echo: Optional[bool] = None
+    echo_pool: Optional[bool] = None
+    enable_from_linting: Optional[bool] = None
+    future: bool = True
+    hide_parameters: Optional[bool] = None
+    isolation_level: Optional[IsolationLevel] = None
+    json_deserializer: Callable[[str], Any] = loads
+    json_serializer: Callable[[Any], str] = serializer
+    label_length: Optional[int] = None
+    listeners: Any = None
+    logging_level: Optional[Union[int, str]] = None
+    logging_name: Optional[str] = None
+    max_identifier_length: Optional[int] = None
+    max_overflow: Optional[int] = None
+    module: Any = None
+    paramstyle: Optional[Literal["qmark", "numeric", "named", "format", "pyformat"]] = None
+    plugins: Optional[List[str]] = None
+    pool: Optional[Pool] = None
+    pool_logging_name: Optional[str] = None
+    pool_pre_ping: Optional[bool] = None
+    pool_recycle: Optional[int] = None
+    pool_reset_on_return: Optional[Literal["rollback", "commit"]] = None
+    pool_size: Optional[int] = None
+    pool_timeout: Optional[int] = None
+    pool_use_lifo: Optional[bool] = None
+    poolclass: Optional[Type[Pool]] = None
+    query_cache_size: Optional[int] = None
+    strategy: Optional[str] = None
+
+
+class SQLAlchemyConfig(GenericModel, Generic[T]):
+    class Config(BaseConfig):
+        arbitrary_types_allowed = True
+
+    connection_string: str
+    middleware_class: Type[SQLAlchemySessionMiddleware] = SQLAlchemySessionMiddleware
+    create_engine_callable: Callable[[str], Union[Engine, FutureEngine]] = create_engine
+    create_async_engine_callable: Callable[[str], AsyncEngine] = create_async_engine
+    create_async_engine: bool = True
+    session_class: Optional[Type[T]] = None
+    session_scope_key: str = "db_session"
```

Comment:
I think we need a `session_maker_callable` and `async_session_maker_callable`

Reply:
it just one callable - its the sessionmaker, which gets a different session types.

Reply:
I addded:

```python
    session_maker: Type[sessionmaker] = sessionmaker
```